### ISSUE TITLE
Some question about the threshold update 

### ISSUE THREAD

**Issue by jianzhengming** [AtmaHou/FewShotMultiLabel#4]
In Line 45-48 of text_classifier.py, 
   filtered_logits = logits - threshold  # For each pos, > 0 for positive tag, < 0 for negative tag.
   multi_hot_target = self.create_multi_hot(targets, label_num=logits.shape[-1])
   loss = self.criterion(filtered_logits, multi_hot_target)

Such subtraction probably causes some logits < 0, it cannot work for the following the loss computation. And each logit subtract the same threshold, it only increases the loss. During training, does the increased loss update the threshold by backwards()?

**Comment by AtmaHou** [comment]
Hi, 
We use the nn.MultiLabelSoftMarginLoss() as loss criterion, which actually takes input logits ranging from -inf and +inf and applies the sigmoid function over logits to ensure a value range of (0, 1). 

See the PyTorch doc as a reference:
https://pytorch.org/docs/stable/generated/torch.nn.MultiLabelSoftMarginLoss.html#multilabelsoftmarginloss
